packages/mui-joy/src/AutocompleteListbox/AutocompleteListbox.tsx

```diff
@@ -0,0 +1,170 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import { OverridableComponent } from '@mui/types';
+import { unstable_capitalize as capitalize } from '@mui/utils';
+import { useSlotProps } from '@mui/base/utils';
+import composeClasses from '@mui/base/composeClasses';
+import { ListRoot } from '../List/List';
+import { styled, useThemeProps } from '../styles';
+import { getAutocompleteListboxUtilityClass } from './autocompleteListboxClasses';
+import {
+  AutocompleteListboxOwnerState,
+  AutocompleteListboxTypeMap,
+} from './AutocompleteListboxProps';
+import listItemClasses from '../ListItem/listItemClasses';
+import listClasses from '../List/listClasses';
+import { scopedVariables } from '../List/ListProvider';
+
+const useUtilityClasses = (ownerState: AutocompleteListboxOwnerState) => {
+  const { variant, color, size } = ownerState;
+  const slots = {
+    root: [
+      'root',
+      variant && `variant${capitalize(variant)}`,
+      color && `color${capitalize(color)}`,
+      size && `size${capitalize(size)}`,
+    ],
+  };
+
+  return composeClasses(slots, getAutocompleteListboxUtilityClass, {});
+};
+
+export const AutocompleteListboxRoot = styled(ListRoot, {
```

Comment:
Why not the `List` component itself?

Reply:
Because `AutocompleteListboxRoot` needs to reuse only the styles of the List without having the `List-root` classname.

Reply:
Aaaah, nice trick I see what you did there :D